### PR TITLE
default to not using the quadratic objective with SCS

### LINKED ISSUE #141: Runing the example "ReLU Layers" encounts the following errors
ValueError                                Traceback (most recent call last)
Cell In[6], line 6
      4 l = torch.nn.MSELoss()(net(X), Y)
      5 print (l.item())
----> 6 l.backward()
      7 opt.step()

File D:\anaconda3\envs\pytorch\lib\site-packages\torch\_tensor.py:487, in Tensor.backward(self, gradient, retain_graph, create_graph, inputs)
    477 if has_torch_function_unary(self):
    478     return handle_torch_function(
    479         Tensor.backward,
    480         (self,),
   (...)
    485         inputs=inputs,
    486     )
--> 487 torch.autograd.backward(
    488     self, gradient, retain_graph, create_graph, inputs=inputs
    489 )

File D:\anaconda3\envs\pytorch\lib\site-packages\torch\autograd\__init__.py:197, in backward(tensors, grad_tensors, retain_graph, create_graph, grad_variables, inputs)
    192     retain_graph = create_graph
    194 # The reason we repeat same the comment below is that
    195 # some Python versions print out the first line of a multi-line function
    196 # calls in the traceback and some print out the last line
--> 197 Variable._execution_engine.run_backward(  # Calls into the C++ engine to run the backward pass
    198     tensors, grad_tensors_, retain_graph, create_graph, inputs,
    199     allow_unreachable=True, accumulate_grad=True)

File D:\anaconda3\envs\pytorch\lib\site-packages\torch\autograd\function.py:267, in BackwardCFunction.apply(self, *args)
    263     raise RuntimeError("Implementing both 'backward' and 'vjp' for a custom "
    264                        "Function is not allowed. You should only implement one "
    265                        "of them.")
    266 user_fn = vjp_fn if vjp_fn is not Function.vjp else backward_fn
--> 267 return user_fn(self, *args)

File D:\anaconda3\envs\pytorch\lib\site-packages\cvxpylayers\torch\cvxpylayer.py:344, in _CvxpyLayerFn.<locals>._CvxpyLayerFnFn.backward(ctx, *dvars)
    342 grad = [[] for _ in range(len(param_ids))]
    343 for i in range(ctx.batch_size):
--> 344     del_param_dict = compiler.apply_param_jac(
    345         dcs[i], -dAs[i], dbs[i])
    346     for j, pid in enumerate(param_ids):
    347         grad[j] += [to_torch(del_param_dict[pid],
    348                              ctx.dtype, ctx.device).unsqueeze(0)]

File D:\anaconda3\envs\pytorch\lib\site-packages\cvxpy\reductions\dcp2cone\cone_matrix_stuffing.py:208, in ParamConeProg.apply_param_jac(self, delc, delA, delb, active_params)
    200 """Multiplies by Jacobian of parameter mapping.
    201 
    202 Assumes delA is sparse.
   (...)
    205     A dictionary param.id -> dparam
    206 """
    207 if self.P is not None:
--> 208     raise ValueError("Can't apply Jacobian with a quadratic objective.")
    210 if active_params is None:
    211     active_params = {p.id for p in self.parameters}

ValueError: Can't apply Jacobian with a quadratic objective.

### PR DESCRIPTION
**Pull request by bamos**
Resolves #141

### CODE REVIEW

File: cvxpylayers/torch/cvxpylayer.py [hunk 1]
```diff
@@ -106,12 +106,13 @@ def __init__(self, problem, parameters, variables, gp=False):
                     raise ValueError("An initial value for each parameter is "
                                      "required when gp=True.")
             data, solving_chain, _ = problem.get_problem_data(
-                solver=cp.SCS, gp=True)
+                solver=cp.SCS, gp=True, solver_opts={'use_quad_obj': False})
```
Comment (SteveDiamond) [inline cvxpylayers/torch/cvxpylayer.py:109]: it's just ``use_quad_obj = False``, no solver_opts.

Comment (bamos) [reply cvxpylayers/torch/cvxpylayer.py:109]: Are you sure? Setting it in `solver_opts` fixes the linked issue and setting `use_quad_obj=False` raises:

```python
  File ~/repos/cvxpylayers/cvxpylayers/torch/cvxpylayer.py:115 in __init__
    data, _, _ = problem.get_problem_data(
TypeError: get_problem_data() got an unexpected keyword argument 'use_quad_obj'

> /private/home/bda/repos/cvxpylayers/cvxpylayers/torch/cvxpylayer.py(115)__init__()
    113             self.param_ids = [p.id for p in self.compiler.parameters]
    114         else:
--> 115             data, _, _ = problem.get_problem_data(
    116                 # solver=cp.SCS, solver_opts={'use_quad_obj': False})
    117                 solver=cp.SCS, use_quad_obj=False)
```

Comment (SteveDiamond) [reply cvxpylayers/torch/cvxpylayer.py:109]: Ah sorry, ``get_problem_data`` is different than solve. The ``solver_opts`` way is the better way to do it, but it's not in ``solve`` for legacy reasons.

**Review by SteveDiamond — APPROVED** [review]

**Comment by bamos** [comment]
Hi @akshayka @sbarratt  - can one of you approve this and merge it in?

**Pull request merged**